src/main/java/webserver/RequestEntity.java

```diff
@@ -0,0 +1,96 @@
+package webserver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import util.HttpRequestUtils;
+import util.IOUtils;
+
+public class RequestEntity {
+	private final String method;
+	private final String path;
+	private final Map<String, String> body;
+	private final Map<String, String> queryStrings;
+	private final Map<String, String> cookies;
+
+	public RequestEntity(String method, String path, Map<String, String> body, Map<String, String> queryStrings, Map<String, String> cookies) {
+		this.method = method;
+		this.path = path;
+		this.body = body;
+		this.queryStrings = queryStrings;
+		this.cookies = cookies;
+	}
+
+	public static RequestEntity from(BufferedReader bufferedReader) throws IOException {
+		List<String> httpRequestContents = HttpRequestUtils.getHttpRequestContents(bufferedReader);
+
+		String[] startLines = httpRequestContents.get(0)
+			.split(" ");
+
+		String method = startLines[0];
+		String requestUrl = startLines[1];
+
+		int queryStringStartIndex = requestUrl.indexOf("?");
+
+		String path;
+		Map<String, String> queryStrings;
+
+		if (queryStringStartIndex == -1) {
+			path = requestUrl;
+			queryStrings = Collections.emptyMap();
+		} else {
+			path = requestUrl.substring(0, queryStringStartIndex);
+			String wholeQueryString = requestUrl.substring(queryStringStartIndex);
+			queryStrings = HttpRequestUtils.parseQueryString(wholeQueryString);
+		}
```

Comment:
`queryStrings`를 먼저 초기화해도 좋을 것 같네요

```suggestion
String path;
Map<String, String> queryStrings = new HashMap<>();

if (queryStringStartIndex == -1) {
	path = requestUrl;
	queryStrings = Collections.emptyMap();
} else {
	path = requestUrl.substring(0, queryStringStartIndex);
	String wholeQueryString = requestUrl.substring(queryStringStartIndex);
	queryStrings.putAll(HttpRequestUtils.parseQueryString(wholeQueryString));
}
```

Reply:
queryStrings의 경우 굳이 분기처리를 하지 않아도 됐었네요.